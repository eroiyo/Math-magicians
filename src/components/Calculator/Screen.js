import React, { useState, useEffect } from 'react';

const Screen = (props) => {
  const [state, changeState] = useState(props);

  useEffect(() => {
    changeState(props);
  }, [props]);
  return (
    <div className="screen">{state.total}</div>
  );
};

Screen.displayName = 'Screen';

Screen.defaultProps = {
  total: '0',

};

export default Screen;
