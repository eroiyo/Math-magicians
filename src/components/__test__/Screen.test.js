import React from 'react';
import Screen from '../Calculator/Screen';
import { render } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import renderer from 'react-test-renderer';

describe ("The screen display data from the props", ()=>{
    it('the screen renders correctly', () => {
        const tree = renderer.create(<Screen />);
    
        expect(tree).toMatchSnapshot();
      });

    it('trying with 5', ()=> {
      const screen = render(<Screen total = "5" />);
      const trueScreen =  screen.getByTestId('screen')
      expect(trueScreen.textContent).toBe('5');
    })
})