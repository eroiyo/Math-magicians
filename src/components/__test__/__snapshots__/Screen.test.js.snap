// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The screen display data from the props the screen renders correctly 1`] = `
<div
  className="screen"
  data-testid="screen"
>
  0
</div>
`;
